warehouse/query-core/src/main/java/datawave/query/jexl/visitors/ParallelIndexExpansion.java

```diff
@@ -487,24 +520,71 @@ private boolean computeExpansionForSubtree(JexlNode node, Join join, Map<JexlNod
         return expand;
     }
     
-    protected boolean isIdentifier(JexlNode node) {
-        return node instanceof ASTERNode;
+    /**
+     * Given a JexlNode, get all grandchildren which follow a path from ASTReference to ASTIdentifier, returning true if the image of the ASTIdentifier is equal
+     * to {@link Constants#ANY_FIELD}
+     *
+     * @param node
+     *            The starting node to check
+     * @return
+     */
+    protected boolean hasUnfieldedIdentifier(JexlNode node) {
+        if (null == node || 2 != node.jjtGetNumChildren()) {
+            return false;
+        }
+        
+        for (int i = 0; i < node.jjtGetNumChildren(); i++) {
+            JexlNode child = node.jjtGetChild(i);
+            
+            if (null != child && child instanceof ASTReference) {
+                for (int j = 0; j < child.jjtGetNumChildren(); j++) {
+                    JexlNode grandChild = child.jjtGetChild(j);
+                    
+                    // If the grandchild and its image is non-null and equal to
+                    // the any-field identifier
+                    if (null != grandChild && grandChild instanceof ASTIdentifier && null != grandChild.image && Constants.ANY_FIELD.equals(grandChild.image)) {
+                        return true;
+                    }
+                }
+            }
+        }
+        
+        return false;
+    }
+    
+    // This flag should keep track of whether we are in a negated portion of the tree.
+    protected boolean negated = false;
+    
+    protected void negate() {
+        this.negated = !this.negated;
+    }
+    
+    protected void revertNegation() {
+        this.negated = !this.negated;
+    }
```

Comment:
combine to single method toggleNegation()

Reply:
will do